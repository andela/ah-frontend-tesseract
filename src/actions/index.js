/* eslint-disable camelcase,no-return-await,no-shadow */
import { axiosInstance } from '../globals';
import {
  REQUEST_PASSWORD_RESET, CONFIRM_PASSWORD_RESET, PROCESSING, FAILED,
} from './types';

export * from "./authentication";
export * from "./articles"
export const requestReset = payload => ({
  type: REQUEST_PASSWORD_RESET,
  payload,
});


export const confirmReset = payload => ({
  type: CONFIRM_PASSWORD_RESET,
  payload,
});

export const processingAction = payload => ({
  type: PROCESSING,
  payload,
});

export const failedAction = payload => ({
  type: FAILED,
  payload,
});

const CatchError = (dispatch, error) => {
    try{
         dispatch(failedAction(error.response.data));
    }catch (e) {
        
    }
    dispatch(processingAction(false));
};

export const handleRequestReset = user_email => async (dispatch) => {
  dispatch(processingAction(true));
  await (
    axiosInstance.post('/password-reset/', user_email)
      .then((response) => {
        localStorage.setItem('TOKEN', response.data.token);

        dispatch(requestReset(response.data));
        dispatch(processingAction(false));
      })
      .catch((error) => {
        try {
          CatchError(dispatch, error);
        } catch (error) {
          console.log('check your internet connection');
        }
      })
  );
};

export const handlePasswordReset = user_password => async (dispatch) => {
  dispatch(processingAction(true));
  const headers = {
    'Content-Type': 'application/json',
    Authorization: localStorage.getItem('TOKEN'),
  };
  return await (
    axiosInstance.put('/password-reset/done', user_password, { headers })
      .then((response) => {
        dispatch(confirmReset(response.data));
        dispatch(processingAction(false));
      })
      .catch((error) => {
        CatchError(dispatch, error);
      })
  );
};
