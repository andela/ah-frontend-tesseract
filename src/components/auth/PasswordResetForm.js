import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {handlePasswordReset} from "../../actions";
import {connect} from "react-redux";
import { validate } from './formValidation';

export class PasswordResetForm extends Component {
        state = { password:'',
                confirmPassword: '' 
                }
    
    handleChange = (e) =>{
        this.setState({[e.target.name]: e.target.value });
    };
    handleSubmit = (event) =>{
        
        event.preventDefault();
            if (validate(this.formEl)){
                if (this.state.password === this.state.confirmPassword){
                    this.props.handlePasswordReset({'user':{...this.state}})

                }else{
                    const elem = this.formEl;
                    const errorLabel = elem.parentNode.querySelector('.invalid-feedback');
                    errorLabel.textContent = 'Your Passwords do not match';
                }
            }     
    };
    
    render() { 
        return ( 
                <div className="container z-depth-2">
                    <h2 className="blue-grey-text center subheader">Enter your new password</h2>
                    <div className="row">
                        <form onSubmit ={this.handleSubmit} ref={form => this.formEl = form} className="col s8 m7 l6 offset-s2 offset-m3 offset-l3" noValidate>
                            <div className="input-field">
                                <i className="material-icons prefix">lock</i>
                                <input id="password" name ='password' value= {this.state.password} onChange = {this.handleChange} type="password" className="validate" placeholder="new password" minLength={8}pattern="(?=.*\d)(?=.*[a-z]).{8,}" required />
                                <label htmlFor="password">Enter your new password</label>
                                <div className={"invalid-feedback"} style={{color: 'red'}}/>
                            </div>
                            <div className="input-field">
                            <i className="material-icons prefix">lock</i>
                                <input type="password" id="comfirmPassword" name="confirmPassword" value = {this.state.confirmPassword} onChange={this.handleChange} className="validate" placeholder="comfirm new password" minLength={8} pattern="(?=.*\d)(?=.*[a-z]).{8,}" required/>
                                <label htmlFor="comfirmPassword">Comfirm your new password</label>
                                <div className={"invalid-feedback"} style={{color: 'red'}}/>  
                            </div>
                            <ResetButton processing = {this.props.processing} handleSubmit = {this.handleSubmit} />
                            
                        </form>
                    </div>
                    <p style={{color: 'green'}}>{this.props.msg.message}</p>
                </div>);}
}
 
export const ResetButton = (props) => {
    return ( <div>
                <div>
                    <strong className={"left"}>Note:</strong>
                    <br/>
                    <small className={"left form-text text muted"}>
                        The password must be at least 8 characters long and contain numbers and letters
                    </small>
                </div>
                <div className="col s12 social" style={{paddingTop: "30px"}}>
                    {props.processing ? <button type="submit" className={"row waves-effect waves-light btn disabled"}>PROCESSING ...</button>
                        :
                        <button type = "submit" className="btn blue-grey waves-effect darken-4" onClick={props.handleSubmit}>Reset Password</button>
                        }
                </div>  
        </div> );
}
 

handlePasswordReset.propTypes = {
    handleRequestReset: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
        msg: state.confirmPasswordReset.message,
        processing: state.confirmPasswordReset.processing,
        error: state.confirmPasswordReset.error
   
});
export default connect(mapStateToProps, {handlePasswordReset})(PasswordResetForm);

