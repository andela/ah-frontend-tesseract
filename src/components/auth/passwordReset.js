import React from 'react'
import PropTypes from 'prop-types';
import { handleRequestReset } from "../../actions";
import { connect } from "react-redux";
import { validate } from './formValidation';
import M from 'materialize-css'


export class PasswordReset extends React.Component {

    state = { email: '' };

    handleChange = (e) => {
        e.preventDefault();
        this.setState({ [e.target.name]: e.target.value });
    };
    handleSubmit = async event => {
        const host = window.location.host + "/redirect_passwordReset"
        event.preventDefault();
        if (validate(this.formEl) ) {
           await this.props.handleRequestReset({
                'email': this.state.email,
                "callback_url": host
            });
            
            if (!this.props.processing && this.props.error){
                M.toast({html: 'This email address is not registered, please sign up ', classes: 'red darken-4', });
                }
                else if (!this.props.processing && this.props.msg.message){
                    M.toast({html: 'Please check your email for your password reset link', classes: 'green darken-4', });
            }         
        }
        
    };
  
    
    render() {
        

        return (<div className="container z-depth-2">
            <h2 className="blue-grey-text center subheader">Forgot your password ?</h2>
            <div className="row">
                <form id = "request-form" ref={form => this.formEl = form} className="col s8 m7 l6 offset-s2 offset-m3 offset-l3" noValidate>
                    <div className="input-field">
                        <i className="material-icons prefix" >email</i>
                        <input id="email" name="email" value={this.state.email} onChange={this.handleChange} type="email" className="validate" placeholder="email" required />
                        <label htmlFor="email">Enter your email</label>
                        <div className={"invalid-feedback"} style={{ color: 'red' }} />
                    </div>
                    <Button processing= {this.props.processing} handleSubmit= {this.handleSubmit}/>
                </form>
            </div>

        </div>
        );

    }
}
export const Button = (props) => {
    return ( <div>
        <div>
            <strong className={"left"}>Note:</strong>
            <br />
            <small className={" left form-text text muted"}>The email must be in the correct format e.g abc@gmail.com</small>
        </div>
        <div className="col s12 social" style={{ paddingTop: "30px" }}>
            {props.processing ?
                <button type="submit" className={"row waves-effect waves-light btn disabled"}>PROCESSING ...</button>
                :
                <button id = "request-button" type="submit" className="btn blue-grey waves-effect darken-4" onClick={props.handleSubmit}>Request password Resest</button>
            }
        </div>
        </div> );
}
 
PasswordReset.propTypes = {
    handleRequestReset: PropTypes.func.isRequired
};

export const mapStateToProps = (state)=> ({
    msg: state.passwordReset.message,
    processing: state.passwordReset.processing,
    error: state.passwordReset.error

});
export default connect(mapStateToProps, { handleRequestReset })(PasswordReset);

