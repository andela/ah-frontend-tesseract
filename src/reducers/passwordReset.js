/* eslint-disable import/prefer-default-export */
import { REQUEST_PASSWORD_RESET, PROCESSING, FAILED } from '../actions/types';


const initialState = {
  message: '',
  processing: false,
  error: '',
};

export const passwordResetReducer = (state = initialState, action) => {
  switch (action.type) {
    case REQUEST_PASSWORD_RESET:
      return {
        ...state,
        message: action.payload,
        error: '',

      };
    case FAILED:
      return {
        ...state,
        error: action.payload,
      };
    case PROCESSING:
      return {
        ...state,
        processing: action.payload,
      };
    default:
      return { ...state };
  }
};
