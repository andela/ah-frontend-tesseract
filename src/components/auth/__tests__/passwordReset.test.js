import Enzyme, {shallow, render, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { PasswordReset, mapStateToProps } from '../passwordReset';
import  {PasswordResetForm} from "../PasswordResetForm";
import  {mapStateToProps as Mapper} from "../PasswordResetForm";
import React from "react";
import {ResetButton} from "../PasswordResetForm"
import {Button} from "../passwordReset";
import { handlePasswordReset } from '../../../actions';


 Enzyme.configure({adapter: new Adapter() });

const msgProp = {message: " An Email has been sent"};

 describe('PasswordReset', () => {
     const props = {
        handleRequestReset: jest.fn(),
        msg: {message: " An Email has been sent"},
     }

     it('renders well without crashing', function () {
        const event = {
            preventDefault() {},
            target: { value: 'test@gmail.com' }
          };
        const wrapper = shallow(<PasswordReset { ...props}/>);
        wrapper.instance().handleChange = jest.fn()
        wrapper.find('#email').simulate('change', event)
     
    });

    it('maps the state to the props', () =>{
        const state = {
            passwordReset:{
                message: "initial",
                processing: false,
                error: "initial"
            }
        }
        const expectedProp ={
            msg: "initial",
            processing: false,
            error: "initial"
        }
        expect( mapStateToProps(state)).toEqual(expectedProp)

    });

});

describe('PasswordResetForms ', () => {
    it('renders password reset forms without crashing', function () {
       const wrapper = shallow(<PasswordResetForm handleRequestReset= {handlePasswordReset()} msg = {msgProp}/>);
       const passwordFieild = wrapper.find('#password')
        wrapper.instance().handleChange({target:{}})
        
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('#comfirmPassword').text()).toBe("");
   });

   it('maps the state to the props', () =>{
    const state = {
        confirmPasswordReset:{
            message: "initial",
            processing: false,
            error: "initial"
        }
    }
    const expectedProp ={
        msg: "initial",
        processing: false,
        error: "initial"
    }
    expect( Mapper(state)).toEqual(expectedProp)
});
});

describe('PasswordResetForms Button', () => {
    it('renders comfirm password button without crashing', function () {
       const wrapper = shallow(<ResetButton handleSubmit= {jest.fn()} processing= {true}/>);
       expect(wrapper.find('.muted').text()).toBe("The password must be at least 8 characters long and contain numbers and letters");
   });
});

describe('PasswordReset Button', () => {
    it('renders password reset button without crashing', function () {
       const wrapper = shallow(<Button handleSubmit= {handlePasswordReset()} processing= {true}/>);
       expect(wrapper.find('.muted').text()).toBe("The email must be in the correct format e.g abc@gmail.com");
   });
});

