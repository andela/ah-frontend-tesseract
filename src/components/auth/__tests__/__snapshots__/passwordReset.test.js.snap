// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordResetForms  renders password reset forms without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordResetForm
    handleRequestReset={[Function]}
    msg={
      Object {
        "message": " An Email has been sent",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="blue-grey-text center subheader"
        >
          Enter your new password
        </h2>,
        <div
          className="row"
        >
          <form
            className="col s8 m7 l6 offset-s2 offset-m3 offset-l3"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                lock
              </i>
              <input
                className="validate"
                id="password"
                minLength={8}
                name="password"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                placeholder="new password"
                required={true}
                type="password"
                value=""
              />
              <label
                htmlFor="password"
              >
                Enter your new password
              </label>
              <div
                className="invalid-feedback"
                style={
                  Object {
                    "color": "red",
                  }
                }
              />
            </div>
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                lock
              </i>
              <input
                className="validate"
                id="comfirmPassword"
                minLength={8}
                name="confirmPassword"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                placeholder="comfirm new password"
                required={true}
                type="password"
                value=""
              />
              <label
                htmlFor="comfirmPassword"
              >
                Comfirm your new password
              </label>
              <div
                className="invalid-feedback"
                style={
                  Object {
                    "color": "red",
                  }
                }
              />
            </div>
            <ResetButton
              handleSubmit={[Function]}
            />
          </form>
        </div>,
        <p
          style={
            Object {
              "color": "green",
            }
          }
        >
           An Email has been sent
        </p>,
      ],
      "className": "container z-depth-2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter your new password",
          "className": "blue-grey-text center subheader",
        },
        "ref": null,
        "rendered": "Enter your new password",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="col s8 m7 l6 offset-s2 offset-m3 offset-l3"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                lock
              </i>
              <input
                className="validate"
                id="password"
                minLength={8}
                name="password"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                placeholder="new password"
                required={true}
                type="password"
                value=""
              />
              <label
                htmlFor="password"
              >
                Enter your new password
              </label>
              <div
                className="invalid-feedback"
                style={
                  Object {
                    "color": "red",
                  }
                }
              />
            </div>
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                lock
              </i>
              <input
                className="validate"
                id="comfirmPassword"
                minLength={8}
                name="confirmPassword"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                placeholder="comfirm new password"
                required={true}
                type="password"
                value=""
              />
              <label
                htmlFor="comfirmPassword"
              >
                Comfirm your new password
              </label>
              <div
                className="invalid-feedback"
                style={
                  Object {
                    "color": "red",
                  }
                }
              />
            </div>
            <ResetButton
              handleSubmit={[Function]}
            />
          </form>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock
                </i>
                <input
                  className="validate"
                  id="password"
                  minLength={8}
                  name="password"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                  placeholder="new password"
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="password"
                >
                  Enter your new password
                </label>
                <div
                  className="invalid-feedback"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
              </div>,
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock
                </i>
                <input
                  className="validate"
                  id="comfirmPassword"
                  minLength={8}
                  name="confirmPassword"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                  placeholder="comfirm new password"
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="comfirmPassword"
                >
                  Comfirm your new password
                </label>
                <div
                  className="invalid-feedback"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
              </div>,
              <ResetButton
                handleSubmit={[Function]}
              />,
            ],
            "className": "col s8 m7 l6 offset-s2 offset-m3 offset-l3",
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": [Function],
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons prefix"
                  >
                    lock
                  </i>,
                  <input
                    className="validate"
                    id="password"
                    minLength={8}
                    name="password"
                    onChange={[Function]}
                    pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                    placeholder="new password"
                    required={true}
                    type="password"
                    value=""
                  />,
                  <label
                    htmlFor="password"
                  >
                    Enter your new password
                  </label>,
                  <div
                    className="invalid-feedback"
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  />,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "lock",
                    "className": "material-icons prefix",
                  },
                  "ref": null,
                  "rendered": "lock",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "validate",
                    "id": "password",
                    "minLength": 8,
                    "name": "password",
                    "onChange": [Function],
                    "pattern": "(?=.*\\\\d)(?=.*[a-z]).{8,}",
                    "placeholder": "new password",
                    "required": true,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Enter your new password",
                    "htmlFor": "password",
                  },
                  "ref": null,
                  "rendered": "Enter your new password",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "invalid-feedback",
                    "style": Object {
                      "color": "red",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons prefix"
                  >
                    lock
                  </i>,
                  <input
                    className="validate"
                    id="comfirmPassword"
                    minLength={8}
                    name="confirmPassword"
                    onChange={[Function]}
                    pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                    placeholder="comfirm new password"
                    required={true}
                    type="password"
                    value=""
                  />,
                  <label
                    htmlFor="comfirmPassword"
                  >
                    Comfirm your new password
                  </label>,
                  <div
                    className="invalid-feedback"
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  />,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "lock",
                    "className": "material-icons prefix",
                  },
                  "ref": null,
                  "rendered": "lock",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "validate",
                    "id": "comfirmPassword",
                    "minLength": 8,
                    "name": "confirmPassword",
                    "onChange": [Function],
                    "pattern": "(?=.*\\\\d)(?=.*[a-z]).{8,}",
                    "placeholder": "comfirm new password",
                    "required": true,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Comfirm your new password",
                    "htmlFor": "comfirmPassword",
                  },
                  "ref": null,
                  "rendered": "Comfirm your new password",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "invalid-feedback",
                    "style": Object {
                      "color": "red",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleSubmit": [Function],
                "processing": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " An Email has been sent",
          "style": Object {
            "color": "green",
          },
        },
        "ref": null,
        "rendered": " An Email has been sent",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="blue-grey-text center subheader"
          >
            Enter your new password
          </h2>,
          <div
            className="row"
          >
            <form
              className="col s8 m7 l6 offset-s2 offset-m3 offset-l3"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock
                </i>
                <input
                  className="validate"
                  id="password"
                  minLength={8}
                  name="password"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                  placeholder="new password"
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="password"
                >
                  Enter your new password
                </label>
                <div
                  className="invalid-feedback"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
              </div>
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock
                </i>
                <input
                  className="validate"
                  id="comfirmPassword"
                  minLength={8}
                  name="confirmPassword"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                  placeholder="comfirm new password"
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="comfirmPassword"
                >
                  Comfirm your new password
                </label>
                <div
                  className="invalid-feedback"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
              </div>
              <ResetButton
                handleSubmit={[Function]}
              />
            </form>
          </div>,
          <p
            style={
              Object {
                "color": "green",
              }
            }
          >
             An Email has been sent
          </p>,
        ],
        "className": "container z-depth-2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter your new password",
            "className": "blue-grey-text center subheader",
          },
          "ref": null,
          "rendered": "Enter your new password",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className="col s8 m7 l6 offset-s2 offset-m3 offset-l3"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock
                </i>
                <input
                  className="validate"
                  id="password"
                  minLength={8}
                  name="password"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                  placeholder="new password"
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="password"
                >
                  Enter your new password
                </label>
                <div
                  className="invalid-feedback"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
              </div>
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock
                </i>
                <input
                  className="validate"
                  id="comfirmPassword"
                  minLength={8}
                  name="confirmPassword"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                  placeholder="comfirm new password"
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="comfirmPassword"
                >
                  Comfirm your new password
                </label>
                <div
                  className="invalid-feedback"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
              </div>
              <ResetButton
                handleSubmit={[Function]}
              />
            </form>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="input-field"
                >
                  <i
                    className="material-icons prefix"
                  >
                    lock
                  </i>
                  <input
                    className="validate"
                    id="password"
                    minLength={8}
                    name="password"
                    onChange={[Function]}
                    pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                    placeholder="new password"
                    required={true}
                    type="password"
                    value=""
                  />
                  <label
                    htmlFor="password"
                  >
                    Enter your new password
                  </label>
                  <div
                    className="invalid-feedback"
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  />
                </div>,
                <div
                  className="input-field"
                >
                  <i
                    className="material-icons prefix"
                  >
                    lock
                  </i>
                  <input
                    className="validate"
                    id="comfirmPassword"
                    minLength={8}
                    name="confirmPassword"
                    onChange={[Function]}
                    pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                    placeholder="comfirm new password"
                    required={true}
                    type="password"
                    value=""
                  />
                  <label
                    htmlFor="comfirmPassword"
                  >
                    Comfirm your new password
                  </label>
                  <div
                    className="invalid-feedback"
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  />
                </div>,
                <ResetButton
                  handleSubmit={[Function]}
                />,
              ],
              "className": "col s8 m7 l6 offset-s2 offset-m3 offset-l3",
              "noValidate": true,
              "onSubmit": [Function],
            },
            "ref": [Function],
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons prefix"
                    >
                      lock
                    </i>,
                    <input
                      className="validate"
                      id="password"
                      minLength={8}
                      name="password"
                      onChange={[Function]}
                      pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                      placeholder="new password"
                      required={true}
                      type="password"
                      value=""
                    />,
                    <label
                      htmlFor="password"
                    >
                      Enter your new password
                    </label>,
                    <div
                      className="invalid-feedback"
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                    />,
                  ],
                  "className": "input-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "lock",
                      "className": "material-icons prefix",
                    },
                    "ref": null,
                    "rendered": "lock",
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "validate",
                      "id": "password",
                      "minLength": 8,
                      "name": "password",
                      "onChange": [Function],
                      "pattern": "(?=.*\\\\d)(?=.*[a-z]).{8,}",
                      "placeholder": "new password",
                      "required": true,
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Enter your new password",
                      "htmlFor": "password",
                    },
                    "ref": null,
                    "rendered": "Enter your new password",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "invalid-feedback",
                      "style": Object {
                        "color": "red",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons prefix"
                    >
                      lock
                    </i>,
                    <input
                      className="validate"
                      id="comfirmPassword"
                      minLength={8}
                      name="confirmPassword"
                      onChange={[Function]}
                      pattern="(?=.*\\\\d)(?=.*[a-z]).{8,}"
                      placeholder="comfirm new password"
                      required={true}
                      type="password"
                      value=""
                    />,
                    <label
                      htmlFor="comfirmPassword"
                    >
                      Comfirm your new password
                    </label>,
                    <div
                      className="invalid-feedback"
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                    />,
                  ],
                  "className": "input-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "lock",
                      "className": "material-icons prefix",
                    },
                    "ref": null,
                    "rendered": "lock",
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "validate",
                      "id": "comfirmPassword",
                      "minLength": 8,
                      "name": "confirmPassword",
                      "onChange": [Function],
                      "pattern": "(?=.*\\\\d)(?=.*[a-z]).{8,}",
                      "placeholder": "comfirm new password",
                      "required": true,
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Comfirm your new password",
                      "htmlFor": "comfirmPassword",
                    },
                    "ref": null,
                    "rendered": "Comfirm your new password",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "invalid-feedback",
                      "style": Object {
                        "color": "red",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleSubmit": [Function],
                  "processing": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " An Email has been sent",
            "style": Object {
              "color": "green",
            },
          },
          "ref": null,
          "rendered": " An Email has been sent",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
